ray-operator/controllers/ray/utils/validation.go

```diff
@@ -158,6 +159,12 @@ func ValidateRayJobSpec(rayJob *rayv1.RayJob) error {
 }
 
 func ValidateRayServiceSpec(rayService *rayv1.RayService) error {
+	if !reflect.DeepEqual(rayService.Spec.RayClusterSpec, rayv1.RayClusterSpec{}) {
```

Comment:
Is this acceptable that it uses `reflect.DeepEqual` to verify if `RayClusterSpec` or not?

Reply:
Hi @fscnick, can’t we use ValidateRayClusterSpec directly? What will happen if use it directly?

Reply:
```suggestion
	if !reflect.ValueOf(rayService.Spec.RayClusterSpec).IsZero() {
```

Reply:
> Hi @fscnick, can’t we use ValidateRayClusterSpec directly? What will happen if use it directly?

Hi @rueian , it seems that the following test cases implies an empty `RayClusterSpec` is acceptable.

https://github.com/ray-project/kuberay/blob/a721d60413bf4ae6596bc3a80450c436df8b1e27/ray-operator/controllers/ray/utils/validation_test.go#L735-L748

Reply:
However, if trying to apply a yaml without `RayClusterSpec`. It would pass the creation of RayService but fail on creating RayCluster.

<img width="1454" alt="fail" src="https://github.com/user-attachments/assets/11a82b9d-122e-4a26-a53f-d93445a6c167" />

Reply:
@kevin85421 Should we make `RayClusterSpec` to required field for `RayService` CRD?

Reply:
> Should we make RayClusterSpec to required field for RayService CRD?

Wow, I wasn’t aware that RayClusterSpec isn’t a required field in the RayService CRD. It should be required. @fscnick would you mind opening a PR to make it become required? 

Remove `omitempty` and then generate the new CRD YAMLs.

https://github.com/ray-project/kuberay/blob/a721d60413bf4ae6596bc3a80450c436df8b1e27/ray-operator/apis/ray/v1/rayservice_types.go#L78

Reply:
SG. I think we should make this field required first and remove the corresponding tests. And then we can remove this check completely in this PR.

Reply:
> > Should we make RayClusterSpec to required field for RayService CRD?
> 
> Wow, I wasn’t aware that RayClusterSpec isn’t a required field in the RayService CRD. It should be required. @fscnick would you mind opening a PR to make it become required?
> 
> Remove `omitempty` and then generate the new CRD YAMLs.
> 
> https://github.com/ray-project/kuberay/blob/a721d60413bf4ae6596bc3a80450c436df8b1e27/ray-operator/apis/ray/v1/rayservice_types.go#L78

Sure, I'll create another PR to address this.